web/pgadmin/static/js/helpers/Layout/index.jsx

```diff
@@ -470,10 +479,8 @@ export default function Layout({groups, noContextGroups, getLayoutInstance, layo
   const saveTab = (tab) => {
   // 'tab' here is the full TabData object, potentially with 'title', 'content', etc.
   // We only want to save the 'id' and any custom properties needed by loadTab.
-    const savedTab = {
-      id: tab.id,
-    };
-    if (tab.metaData && !BROWSER_PANELS.DEBUGGER_TOOL.includes(tab.id.split('_')[0])) {
+    const saveAppState = prefStore?.getPreferencesForModule('misc')?.save_app_state;
```

Comment:
At this moment, we can safely use `saveAppStateRef.current`

Reply:
Done.